File: lib/models/movie.dart
class Movie {
  final int id;
  final String title;
  final String thumb;
  final String video;

  Movie(this.id, this.title, this.thumb, this.video);

  factory Movie.fromJson(Map<String, dynamic> json) =>
      Movie(json['id'], json['title'], json['thumb'], json['video']);

  @override
  String toString() {
    return "Movie{id: $id, title: $title, thumb: $thumb, video: $video}";
  }
}

File: android/sharedcode/src/commonMain/kotlin/com/vitoksmile/cpmoviemaker/model/Movie.kt
package com.vitoksmile.cpmoviemaker.model

import com.vitoksmile.cpmoviemaker.MovieEntity
import kotlinx.serialization.Serializable

@Serializable
data class Movie(
    override val id: Long,
    override val title: String,
    override val thumb: String,
    override val video: String
) : MovieEntity
